addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -1509,11 +1509,26 @@ export function makeActionManager(env, router = _router) {
             throw new InvalidButtonParamsError("Missing type for doActionButton request");
         }
         if (!isEmbeddedAction && action.embedded_action_ids?.length) {
-            const embeddedActionsOrder = JSON.parse(
-                browser.localStorage.getItem(
-                    `orderEmbedded${action.id}+${params.resId || ""}+${user.userId}`
-                )
-            );
+            const embeddedActionsKey = `${action.id}+${params.resId || ""}`;
+            const embeddedActionsConfig = user.settings.embedded_actions_config_ids;
+            // If there are embedded actions and no config has been found in the settings, we will fetch it from DB
+            // We need to fetch because it's possible that the config from DB was changed while it wasn't in the browser user settings
+            // We then need to keep the browser user settings up to date with the DB
+            if (!(embeddedActionsKey in embeddedActionsConfig)) {
+                // TODO: use orm service
+                const embeddedSetting = await this.orm.call(
+                    "res.users.settings",
+                    "get_embedded_actions_setting",
+                    [user.settings.id, action.id, params.resId]
+                );
+                if (embeddedSetting) {
+                    embeddedActionsConfig[embeddedActionsKey] = embeddedSetting;
+                    user.updateUserSettings("embedded_actions_config_ids", embeddedActionsConfig);
+                }
+            }
```

Comment:
Hum, that's a bit annoying actually.

I think it would be better to not do that extra call because when we will have that extra rpc when the user will start using project and if that user never uses the top bar, he will never have a config for the topbar inside that project and so we will each time do that rpc.

I still think it would be better to have a small limitation. That is, if the user duplicate a project then we will not copy the show top bar. By doing that, when the user will click to show the top bar, we will do that rpc to get the new configs of the user (would be better to refresh the setting to be sure we are up to date for the other eventual projects duplicated).